### PR TITLE
review: alan/no-fork review comments

### PR DESCRIPTION
**Pull request by y0sher**

### CODE REVIEW

File: message/validation/common_checks.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		}
+	}
+
+	// Rule: For a sync committee aggregation duty message, we check if the validator is assigned to it
+	if role == spectypes.RoleSyncCommitteeContribution {
+		period := mv.netCfg.Beacon.EstimatedSyncCommitteePeriodAtEpoch(mv.netCfg.Beacon.EstimatedEpochAtSlot(slot))
+		if mv.dutyStore.SyncCommittee.Duty(period, indices[0]) == nil {
+			return ErrNoDuty
+		}
+	}
```
Comment (y0sher) [inline message/validation/common_checks.go:115]: @MatheusFranco99 @nkryuchkov 

Why we use `indices[0]` here? how do we know this is the correct validator index from the whole committee validator list?

Comment (y0sher) [reply message/validation/common_checks.go:115]: this should probably be checked with either all the indices or the right index that really has duty.

Comment (MatheusFranco99) [reply message/validation/common_checks.go:115]: In this function, I think it's assumed that, if the role is either Proposer or Sync Committee Contribution, the indices list will only have one element. Right @nkryuchkov ?

Comment (MatheusFranco99) [reply message/validation/common_checks.go:115]: It's checked somewhere else but we can add here an extra check to see that the length of the list is 1, for independence, if you like

File: registry/storage/validatorstore.go [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	// Update byValidatorIndex
+	if share.HasBeaconMetadata() {
+		c.byValidatorIndex[share.BeaconMetadata.Index] = share
+	}
+
+	// Update byCommitteeID
+	for _, committee := range c.byCommitteeID {
```
Comment (y0sher) [inline registry/storage/validatorstore.go:309]: like you did in shareAdded -
```suggestion
	for _, committee := range c.byCommitteeID {
	if committee.ID != share.CommitteeID() {
	  continue
	}
```

File: registry/storage/validatorstore.go [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+				break
+			}
+		}
+
+		for i, index := range committee.Indices {
+			if index == share.ValidatorIndex {
+				committee.Indices[i] = share.ValidatorIndex
+				break
+			}
+		}
```
Comment (y0sher) [inline registry/storage/validatorstore.go:322]: @nkryuchkov this doesn't seem right. how can this work? 
if the index is already the correct index from the share why we even need to run this code?
if the index isn't correct - on fresh sync before the metadata update, the index will be 0. then this update code just won't work.

Probably best to set this above, where you found the validator pubkey, and use the same slice index to update the validator index.

File: registry/storage/validatorstore.go (hunk 2 above)
Comment (nkryuchkov) [reply registry/storage/validatorstore.go:309]: https://github.com/ssvlabs/ssv/pull/1495

File: registry/storage/validatorstore.go (hunk 3 above)
Comment (nkryuchkov) [reply registry/storage/validatorstore.go:322]: https://github.com/ssvlabs/ssv/pull/1495

File: message/validation/common_checks.go (hunk 1 above)
Comment (nkryuchkov) [reply message/validation/common_checks.go:115]: https://github.com/ssvlabs/ssv/pull/1496

Comment (y0sher) [reply message/validation/common_checks.go:115]: its not what I meant, I didn't see the part where we are passing the 1 len list. I was now able to track it down to the code at `getCommitteeAndValidatorIndices` where we populate the list with only one validator when the message belongs to a propose duty.

**Pull request merged**